### PR TITLE
Feature/link control improves

### PR DESCRIPTION
**Pull request by joshmarom**

### CODE REVIEW

File: includes/base/element-base.php [hunk 1]
```diff
@@ -423,6 +423,51 @@ public function add_render_attribute( $element, $key = null, $value = null, $ove
 		return $this;
 	}
 
+	/**
+	 * Add link render attributes.
+	 *
+	 * Used to add link tag attributes to a specific HTML element.
+	 *
+	 * The HTML link tag is represented by the element parameter, then you need to
```
Comment (matipojo) [inline includes/base/element-base.php]: You shouldn't use the word `you`...

File: includes/base/element-base.php [hunk 2]
```diff
@@ -423,6 +423,51 @@ public function add_render_attribute( $element, $key = null, $value = null, $ove
...
+	 * pass an array of link settings in the same format they are set by Elementor's URL control.
+	 *
+	 * Example usage:
+	 *
+	 * `$this->add_render_attribute( 'button', $settings['link'] );`
```
Comment (matipojo) [inline includes/base/element-base.php]:
```suggestion
	 * `$this->add_link_attributes( 'button', $settings['link'] );`
```

**Review by matipojo — CHANGES_REQUESTED** [review]

**Review by matipojo — DISMISSED** [review]

File: includes/widgets/button.php [hunk 3]
```diff
@@ -444,15 +444,7 @@ protected function render() {
 
 		if ( ! empty( $settings['link']['url'] ) ) {
 			$this->add_render_attribute( 'button', 'href', $settings['link']['url'] );
```
Comment (bainternet) [inline includes/widgets/button.php]: seems Wrong, as you don't need this anymore

File: includes/widgets/button.php [hunk 4]
```diff
@@ -444,15 +444,7 @@ protected function render() {
...
-			$this->add_render_attribute( 'button', 'class', 'elementor-button-link' );
```
Comment (bainternet) [inline includes/widgets/button.php]: This should still be here

**Review by kobizz — DISMISSED** [review]

File: includes/base/element-base.php [hunk 5]
```diff
@@ -423,6 +423,51 @@ public function add_render_attribute( $element, $key = null, $value = null, $ove
...
+
+		if ( ! empty( $url_control['url'] ) ) {
+			$attributes['href'] = $url_control['url'];
+		}
+
+		if ( ! empty( $url_control['is_external'] ) ) {
+			$attributes['target'] = '_blank';
+		}
+
+		if ( ! empty( $url_control['nofollow'] ) ) {
```
Comment (KingYes) [inline includes/base/element-base.php]: How you think to manage other `rel` attribute?

**Review by KingYes — CHANGES_REQUESTED** [review]

**Comment by bainternet** [comment]
Same for target attribute which now can only be set to `blank`

File: includes/widgets/button.php (hunk 3 above)
Comment (joshmarom) [reply includes/widgets/button.php]: Removed

File: includes/widgets/button.php (hunk 4 above)
Comment (joshmarom) [reply includes/widgets/button.php]: Restored

**Review by kobizz — APPROVED** [review]

**Pull request merged**